logo-ann/benchmark/main.py

```diff
@@ -122,6 +122,10 @@ def pairwise_squared_euclidian_distance(A: np.ndarray) -> torch.Tensor:
     squared_sum = torch.sum(A ** 2.0, axis=1, keepdim=True)
     return squared_sum + squared_sum.T - 2 * dot_product
 
+def cosine_similarity(A: torch.Tensor) -> torch.Tensor:
+    assert len(A.shape) == 2
+    normalized = torch.nn.functional.normalize(A,p=2.0,dim=1)
```

Comment:
Not important here (as it's research code), but it's good to follow PEP8:
`normalized = torch.nn.functional.normalize(A,p=2.0,dim=1)` -> `normalized = torch.nn.functional.normalize(A, p=2.0, dim=1)`